src/components/molecules/EventDisplay/EventDisplay.tsx

```diff
@@ -26,18 +26,17 @@ export const EventDisplay: FC<EventDisplayProps> = ({ event, venue }) => {
   const enterEvent = useCallback(() => {
     const room = venue?.rooms?.find((room) => room.title === event.room);
 
-    const isExternal = isExternalUrl(room.url);
-    if (isExternal) {
-      window.open(room.url, "_blank", "noopener,noreferrer");
-    } else {
-      window.location.href = room.url;
+    if (!room) {
+      window.open(venueInsideUrl(venue.id), "_blank", "noopener,noreferrer");
+      return;
     }
 
     enterRoom(
       user!,
       { [`${venue.name}/${room.title}`]: currentTimeInUnixEpoch },
       profile?.lastSeenIn
     );
+    openRoomUrl(room.url);
```

Comment:
In case we decide/think this will take too long (as discussed on call), could have this first. It should be fine though from memory as the deeper code uses promises.